### PR TITLE
Flutter APP PR

### PR DESCRIPTION
**Pull request by achintya-7**

### CODE REVIEW

**Comment by achintya-7** [comment]
@naveen-egov 
Can you review this please.
Thank you.

File: vehicle-tracker/vehicle_tracker_app/lib/constants.dart [hunk 1]
```diff
@@ -0,0 +1,16 @@
+import 'package:latlong2/latlong.dart';
```
Comment (naveen-egov) [inline vehicle-tracker/vehicle_tracker_app/lib/constants.dart:1]: Create an env file and move this to that env

File: vehicle-tracker/vehicle_tracker_app/lib/features/home/pages/home_page.dart [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+import 'package:vehicle_tracker_app/util/i18n_translations.dart';
+
+import '../../../util/widgets/drawer_widget.dart';
+
+class HomePage extends StatelessWidget {
+  const HomePage({super.key});
+
+  @override
+  Widget build(BuildContext context) {
+    // final placeController = Get.find<PlaceController>();
```
Comment (naveen-egov) [inline vehicle-tracker/vehicle_tracker_app/lib/features/home/pages/home_page.dart:13]: remove un used code

File: vehicle-tracker/vehicle_tracker_app/lib/features/home/pages/home_page.dart [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+    return Scaffold(
+      // * appbar
+      appBar: AppBar(
+        title: Text(AppTranslation.hello.tr),
```
Comment (naveen-egov) [inline vehicle-tracker/vehicle_tracker_app/lib/features/home/pages/home_page.dart:18]: use a valid names

File: vehicle-tracker/vehicle_tracker_app/lib/features/home/pages/map_page.dart [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+
+class _MapPageState extends State<MapPage> {
+  LatLng? markerPosition = Get.arguments['position'];
+
+  @override
+  Widget build(BuildContext context) {
+    return SafeArea(
+      child: Scaffold(
+        appBar: AppBar(
+          title: const Text("Open Street Map Preview"),
```
Comment (naveen-egov) [inline vehicle-tracker/vehicle_tracker_app/lib/features/home/pages/map_page.dart:27]: use i18 key with translators

File: vehicle-tracker/vehicle_tracker_app/lib/features/home/pages/map_page.dart [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+            zoom: 17,
+            maxZoom: 18,
+            minZoom: 1,
+            onPositionChanged: (position, hasGesture) {
+              setState(() {
+                markerPosition = position.center ?? markerPosition;
+              });
+            },
+          ),
+          // nonRotatedChildren: [
```
Comment (naveen-egov) [inline vehicle-tracker/vehicle_tracker_app/lib/features/home/pages/map_page.dart:41]: remove un-used code

File: vehicle-tracker/vehicle_tracker_app/lib/features/home/widgets/home_list_widget.dart [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+    final mapControllers = Get.find<MapControllers>();
+
+    return DigitCard(
+      child: Column(
+        crossAxisAlignment: CrossAxisAlignment.start,
+        children: [
+          Row(
+            children: [
+              const SizedBox(width: 8.0),
+              const Text("Fecal Sludge Management", style: TextStyle(fontSize: 18, fontWeight: FontWeight.bold)),
```
Comment (naveen-egov) [inline vehicle-tracker/vehicle_tracker_app/lib/features/home/widgets/home_list_widget.dart:22]: Move to i 18 constants with translators and don't externally define the style use digit themes. i complete file

File: vehicle-tracker/vehicle_tracker_app/lib/features/home/widgets/map_marker.dart [hunk 7]
```diff
@@ -0,0 +1,24 @@
...
+
+  final LatLng position;
+
+  @override
+  Widget build(BuildContext context) {
+    return MarkerLayer(
+      markers: [
+        Marker(
+          point: position,
+          builder: (context) => const Icon(Icons.place, size: 36, color: Colors.red),
```
Comment (naveen-egov) [inline vehicle-tracker/vehicle_tracker_app/lib/features/home/widgets/map_marker.dart:19]: +1

File: vehicle-tracker/vehicle_tracker_app/lib/features/inbox/controllers/inbox_controllers.dart [hunk 8]
```diff
@@ -0,0 +1,41 @@
...
+    Map<String, dynamic> body = {
+      "RequestInfo": {
+        "apiId": "Rainmaker",
+        "authToken": await SecureStorageService.read() ?? "",
+        "msgId": "1688975382330|en_IN",
+        "plainAccessRequest": {}
+      }
+    };
+
+    String route = "location/v11/boundarys/_search?hierarchyTypeCode=REVENUE&boundaryType=Locality&tenantId=pg.cityb";
```
Comment (naveen-egov) [inline vehicle-tracker/vehicle_tracker_app/lib/features/inbox/controllers/inbox_controllers.dart:32]: endpoints should be in a separate file

File: vehicle-tracker/vehicle_tracker_app/lib/features/inbox/pages/inbox_page.dart [hunk 9]
```diff
@@ -0,0 +1,51 @@
...
+class InboxPage extends StatelessWidget {
+  const InboxPage({super.key});
+
+  @override
+  Widget build(BuildContext context) {
+    InboxControllers inboxControllers = Get.put(InboxControllers());
+    return Scaffold(
+      // * APP BAR
+      appBar: AppBar(
+        title: const Text("Municipal Corporation"),
```
Comment (naveen-egov) [inline vehicle-tracker/vehicle_tracker_app/lib/features/inbox/pages/inbox_page.dart:19]: +1

File: vehicle-tracker/vehicle_tracker_app/lib/features/inbox/pages/inbox_page.dart [hunk 10]
```diff
@@ -0,0 +1,51 @@
...
+
+        return const ScrollableContent(
+          header: InboxHeaderWidget(),
+          crossAxisAlignment: CrossAxisAlignment.stretch,
+          children: [
+            SizedBox(
+              width: double.infinity,
+              child: InboxCardWidget(),
+              // child: DigitCard(
+              //   child: Text(
```
Comment (naveen-egov) [inline vehicle-tracker/vehicle_tracker_app/lib/features/inbox/pages/inbox_page.dart:39]: +1

File: vehicle-tracker/vehicle_tracker_app/lib/features/inbox/widgets/inbox_card_widget.dart [hunk 11]
```diff
@@ -0,0 +1,33 @@
...
+class InboxCardWidget extends StatelessWidget {
+  const InboxCardWidget({super.key});
+
+  @override
+  Widget build(BuildContext context) {
+    return DigitCard(
+      child: Column(
+        children: [
+          _textRow("Application No", "3t4354y7"),
+          _textRow("Applicant Date", "6/7/2023"),
```
Comment (naveen-egov) [inline vehicle-tracker/vehicle_tracker_app/lib/features/inbox/widgets/inbox_card_widget.dart:13]: +1

File: vehicle-tracker/vehicle_tracker_app/lib/features/inbox/widgets/inbox_header_widget.dart [hunk 12]
```diff
@@ -0,0 +1,148 @@
...
+  @override
+  Widget build(BuildContext context) {
+    HeaderControllers headerControllers = Get.find<HeaderControllers>();
+
+    return Padding(
+      padding: const EdgeInsets.symmetric(horizontal: 10.0),
+      child: Column(
+        crossAxisAlignment: CrossAxisAlignment.start,
+        children: [
+          const Text("Home / FSM / Inbox"),
```
Comment (naveen-egov) [inline vehicle-tracker/vehicle_tracker_app/lib/features/inbox/widgets/inbox_header_widget.dart:20]: +1

File: vehicle-tracker/vehicle_tracker_app/lib/features/inbox/widgets/inbox_header_widget.dart [hunk 13]
```diff
@@ -0,0 +1,148 @@
...
+          const Text("INBOX", style: TextStyle(fontSize: 30, fontWeight: FontWeight.bold)),
+          Row(
+            mainAxisAlignment: MainAxisAlignment.spaceBetween,
```
Comment (naveen-egov) [inline vehicle-tracker/vehicle_tracker_app/lib/features/inbox/widgets/inbox_header_widget.dart:23]: create a map and iterate through map

File: vehicle-tracker/vehicle_tracker_app/lib/features/inbox/widgets/inbox_header_widget.dart [hunk 14]
```diff
@@ -0,0 +1,148 @@
...
+
+searchDialogBox(context, HeaderControllers headerControllers) => DigitActionDialog.show(
+      context,
+      widget: Column(
+        crossAxisAlignment: CrossAxisAlignment.start,
+        children: [
+          const Text("Search By", style: TextStyle(fontSize: 20)),
+          DigitTextField(
+            label: "Application No.",
+            controller: headerControllers.applicationNoCtrl,
```
Comment (naveen-egov) [inline vehicle-tracker/vehicle_tracker_app/lib/features/inbox/widgets/inbox_header_widget.dart:62]: +1

File: vehicle-tracker/vehicle_tracker_app/lib/features/inbox/widgets/inbox_header_widget.dart [hunk 15]
```diff
@@ -0,0 +1,148 @@
...
+      widget: Obx(
+        () => Column(
+          crossAxisAlignment: CrossAxisAlignment.start,
+          children: [
+            const Text("Filter By", style: TextStyle(fontSize: 20)),
+            const SizedBox(height: 10),
+            const Text("Locality"),
+            const SizedBox(height: 20),
+            const Text("Status"),
+            DigitCheckboxTile(
```
Comment (naveen-egov) [inline vehicle-tracker/vehicle_tracker_app/lib/features/inbox/widgets/inbox_header_widget.dart:91]: +1

**Review by naveen-egov — CHANGES_REQUESTED** [review]
Added a few comments pls update the code accordingly

File: vehicle-tracker/vehicle_tracker_app/lib/features/home/pages/home_page.dart (hunk 2 above)
Comment (achintya-7) [reply vehicle-tracker/vehicle_tracker_app/lib/features/home/pages/home_page.dart:13]: ok

File: vehicle-tracker/vehicle_tracker_app/lib/features/home/pages/home_page.dart (hunk 3 above)
Comment (achintya-7) [reply vehicle-tracker/vehicle_tracker_app/lib/features/home/pages/home_page.dart:18]: Yes, have added the changes

File: vehicle-tracker/vehicle_tracker_app/lib/features/home/pages/map_page.dart (hunk 4 above)
Comment (achintya-7) [reply vehicle-tracker/vehicle_tracker_app/lib/features/home/pages/map_page.dart:27]: As per the new Figma designs. This map page is not required in the app so I am removing it.

File: vehicle-tracker/vehicle_tracker_app/lib/features/inbox/widgets/inbox_header_widget.dart (hunk 13 above)
Comment (achintya-7) [reply vehicle-tracker/vehicle_tracker_app/lib/features/inbox/widgets/inbox_header_widget.dart:23]: Can you explain a little bit more for this

File: vehicle-tracker/vehicle_tracker_app/lib/features/home/pages/map_page.dart (hunk 5 above)
Comment (achintya-7) [reply vehicle-tracker/vehicle_tracker_app/lib/features/home/pages/map_page.dart:41]: Have removed the map page

File: vehicle-tracker/vehicle_tracker_app/lib/features/home/widgets/map_marker.dart (hunk 7 above)
Comment (achintya-7) [reply vehicle-tracker/vehicle_tracker_app/lib/features/home/widgets/map_marker.dart]: removed

File: vehicle-tracker/vehicle_tracker_app/lib/constants.dart (hunk 1 above)
Comment (achintya-7) [reply vehicle-tracker/vehicle_tracker_app/lib/constants.dart:1]: ok

File: vehicle-tracker/vehicle_tracker_app/lib/features/home/widgets/home_list_widget.dart (hunk 6 above)
Comment (achintya-7) [reply vehicle-tracker/vehicle_tracker_app/lib/features/home/widgets/home_list_widget.dart:22]: alright

File: vehicle-tracker/vehicle_tracker_app/.env [hunk 16]
```diff
@@ -1,5 +1,6 @@
 API_URL=https://uat.digit.org/egov-location/
 LOCALIZATION_API_URL=https://health-qa.digit.org/localization/messages/v1/_search?
+MDMS_URL=https://raw.githubusercontent.com/egovernments/egov-mdms-data/UNIFIED-DEV/data/pb/common-masters/StateInfo.json
```
Comment (naveen-egov) [inline vehicle-tracker/vehicle_tracker_app/.env:3]: This should not be a json path :
Fix : {baseURL}/egov-mdms-service/v1/_search?tenantId=pb
and pass the modules to get the data

**Review by naveen-egov — CHANGES_REQUESTED** [review]

File: vehicle-tracker/vehicle_tracker_app/.env (hunk 16 above)
Comment (achintya-7) [reply vehicle-tracker/vehicle_tracker_app/.env:3]: Will require the exact url, I am getting this response

```
{
  "ResponseInfo": null,
  "Errors": [
    {
      "code": "NotNull.mdmsCriteriaReq.mdmsCriteria",
      "message": "must not be null",
      "description": null,
      "params": null
    },
    {
      "code": "NotNull.mdmsCriteriaReq.requestInfo",
      "message": "must not be null",
      "description": null,
      "params": null
    }
  ]
}
```

**Pull request closed without merging**